core/src/main/java/com/google/cloud/sql/core/CoreSocketFactory.java

```diff
@@ -211,15 +212,29 @@ public static Socket connect(Properties props, String unixPathSuffix)
     }
 
     final List<String> ipTypes = listIpTypes(props.getProperty("ipTypes", DEFAULT_IP_TYPES));
-    return getInstance().createSslSocket(csqlInstanceName, ipTypes, enableIamAuth);
+    if (enableIamAuth) {
+      return getInstance().createSslSocket(csqlInstanceName, ipTypes, AuthType.IAM);
+    }
+    return getInstance().createSslSocket(csqlInstanceName, ipTypes, AuthType.PASSWORD);
   }
 
   /**
    * Returns data that can be used to establish Cloud SQL SSL connection.
    */
   public static SslData getSslData(String csqlInstanceName, boolean enableIamAuth)
       throws IOException {
-    return getInstance().getCloudSqlInstance(csqlInstanceName, enableIamAuth).getSslData();
+    if (enableIamAuth) {
+      return getInstance().getCloudSqlInstance(csqlInstanceName, AuthType.IAM).getSslData();
+    }
+    return getInstance().getCloudSqlInstance(csqlInstanceName, AuthType.PASSWORD).getSslData();
+  }
+
+  /**
+   * Returns data that can be used to establish Cloud SQL SSL connection.
+   */
+  public static SslData getSslData(String csqlInstanceName, AuthType authType)
```

Comment:
Does this need to be public?

Reply:
It does because the Connector implementation will use this instead of the method which uses the boolean parameter.